nn_sdwan/datadog_checks/nn_sdwan/check.py

```diff
@@ -0,0 +1,137 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+# import net
+
+import platform
+
+from datadog_checks.base import AgentCheck, ConfigurationError
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+from .net_api import vManageApi
+
+
+class NnSdwanCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = 'nn_sdwan'
+
+    def __init__(self, name, init_config, instances):
+        super(NnSdwanCheck, self).__init__(name, init_config, instances)
+
+        # Validate instance config fields are present
+        for field in 'hostname username password protocol'.split():
+            if not self.instance.get(field):
+                raise ConfigurationError(f'Configuration error: "{field}" not present.')
+
+        self.vmanage_api = vManageApi(instance=instances[0])
+
+    def check(self, _):
+        # Handle service check (availablity / reachability)
+        if not self.pingable(self.instance.get('hostname')):
+            self.service_check('nn_sdwan.sdwan_controller.online', self.CRITICAL, message='Host unreachable.')
+            return
+
+        try:
+            self.vmanage_api.get_device_info()
+            # vManage is online and responding
+            self.service_check('nn_sdwan.sdwan_controller.online', self.OK)
+        except Exception:
+            self.service_check(
+                'nn_sdwan.sdwan_controller.online', self.WARNING, message='vManage API is not responding.'
+            )
+            return
+
+        # Get metrics
+        self.get_control_status()
+        self.get_top_application_stats()
+        self.get_application_aware_routing()
+        self.get_connection_summary_stats()
+        self.get_certificate_summary()
+        self.get_reboot_count()
+        self.get_vmanage_count()
+        self.get_site_health()
+        self.get_transport_interface()
+        self.get_wan_edge_health()
+        self.get_wan_edge_inventory()
```

Comment:
Would it make sense to try wrapping this in a try for error handling? Incase any of the function fails, then at least we don't throw an exception.